tests/io/test_data_catalog.py

```diff
@@ -867,38 +895,13 @@ def test_sorting_order_with_default_and_other_dataset_through_extra_pattern(
             "{namespace}_{dataset}",
             "{dataset}s",
             "{default}",
-            "{user_default}",
         ]
         assert (
             list(catalog_with_default._dataset_patterns.keys()) == sorted_keys_expected
         )
 
-    def test_runner_default_overwrites_user_default(
-        self, config_with_dataset_factories_only_patterns
-    ):
-        """Check that the runner default overwrites the user default."""
-        catalog = DataCatalog.from_config(**config_with_dataset_factories_only_patterns)
-        assert catalog._dataset_patterns["{default}"] == {
-            "filepath": "data/01_raw/{default}.csv",
-            "type": "pandas.CSVDataset",
-        }
-
-        extra_dataset_patterns = {
-            "{default}": {"type": "MemoryDataset"},
-            "{another}#csv": {
-                "type": "pandas.CSVDataset",
-                "filepath": "data/{another}.csv",
-            },
-        }
-        catalog_with_runner_default = catalog.shallow_copy(
-            extra_dataset_patterns=extra_dataset_patterns
-        )
-        assert catalog_with_runner_default._dataset_patterns["{default}"] == {
-            "type": "MemoryDataset"
-        }
-
-    def test_user_default_overwrites_runner_default_alphabetically(self):
-        """Check that the runner default overwrites the user default if earlier in alphabet."""
+    def test_user_default_overwrites_runner_default(self):
```

Comment:
I think we should make this behaviour more explicit by documenting it and perhaps also giving a warning.

Reply:
We already have a warning when a catch-all pattern is being used for a dataset, do you mean another one up top when we set the `DataCatalog._default_pattern`? 

I feel like it would be too many warnings for a simple run where the user is using a feature as intended? Maybe one warning in `from_config()` instead of the warning that shows everytime you fetch a dataset that matches the default pattern?

Reply:
Fair point. Maybe specify in the error that there can only be one catch-all pattern across the catalog and runner?

Reply:
I think it's clear now after your changes to the error message 👍